src/libraries/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Certificate2Collection.cs

```diff
@@ -413,5 +414,195 @@ public void ImportFromPem(ReadOnlySpan<char> certPem)
+                    Debug.Fail("Pre-allocated buffer was not the correct size.");
+                    throw new CryptographicException();
+                }
+            });
+        }
+
+        /// <summary>
+        /// Attempts to export the public X.509 certificates, encoded as PEM.
+        /// </summary>
+        /// <param name="destination">The buffer to receive the PEM encoded certificates.</param>
+        /// <param name="charsWritten">When this method returns, the total number of characters written to <paramref name="destination" />.</param>
+        /// <returns>
+        ///   <see langword="true"/> if <paramref name="destination"/> was large enough to receive the encoded PEMs;
+        ///   otherwise, <see langword="false" />.
+        /// </returns>
+        /// <exception cref="CryptographicException">
+        /// A certificate is corrupt, in an invalid state, or could not be exported
+        /// to PEM.
+        /// </exception>
+        /// <remarks>
+        /// <p>
+        ///   A PEM-encoded X.509 certificate collection will contain certificates
+        ///   where each certificate begins with <c>-----BEGIN CERTIFICATE-----</c>
+        ///   and ends with <c>-----END CERTIFICATE-----</c>, with the base64 encoded DER
+        ///   contents of the certificate between the PEM boundaries. Each certificate is
+        ///   separated by a single line-feed character.
+        /// </p>
+        /// <p>
+        ///   Certificates are encoded according to the IETF RFC 7468 &quot;strict&quot;
+        ///   encoding rules.
+        /// </p>
+        /// </remarks>
+        public bool TryExportCertificatePems(Span<char> destination, out int charsWritten)
+        {
+            Span<char> buffer = destination;
+            int written = 0;
+
+            for (int i = 0; i < Count; i++)
+            {
+                ReadOnlyMemory<byte> certData = this[i].RawDataMemory;
+                int certSize = PemEncoding.GetEncodedSize(PemLabels.X509Certificate.Length, certData.Length);
+
+                if (buffer.Length < certSize)
+                {
+                    charsWritten = 0;
+                    return false;
+                }
+
+                if (!PemEncoding.TryWrite(PemLabels.X509Certificate, certData.Span, buffer, out int certWritten) ||
+                    certWritten != certSize)
+                {
+                    Debug.Fail("Presized buffer is too small or did not write the correct amount.");
+                    throw new CryptographicException();
+                }
+
+                buffer = buffer.Slice(certWritten);
+                written += certWritten;
+
+                // write a new line if not the last certificate.
+                if (i < Count - 1)
+                {
+                    if (buffer.IsEmpty)
+                    {
+                        charsWritten = 0;
+                        return false;
+                    }
+
+                    buffer[0] = '\n';
```

Comment:
Please add a comment here that it's Unix newlines, vs platform newlines, because that matches what PemEncoding.TryWrite is going to do, and we don't want to mix modes.